pyethapp/jsonrpc.py

```diff
@@ -839,10 +840,10 @@ class Filter(object):
     :ivar new_logs: same as :attr:`logs`, but is reset at every access
     """
 
-    def __init__(self, chain, blocks=[], addresses=None, topics=None,
+    def __init__(self, chain, blocks, addresses=None, topics=None,
```

Comment:
looking at the code i actually think the signature should be:
`def __init__(self, chain, blocks=set(), addresses=set() topics=set(), pending=False, latest=False)`

combined with appropriate changes in the code following.

BUT: it also depends on how the code is called. i'd rather not change anything here without proper unit tests.